### PR TITLE
Sync Lock: prevent setting explicit leader to stopped decks

### PR DESCRIPTION
**Pull request by ywwg**
adjusted tests to match the new behavior

### CODE REVIEW

File: res/skins/LateNight/deck_rate_controls.xml [hunk 1]
```diff
@@ -190,7 +190,7 @@
                 <SetVariable name="Size">52f,26f</SetVariable>
                 <SetVariable name="btn_format">wide</SetVariable>
                 <SetVariable name="ConfigKey"><Variable name="group"/>,sync_enabled</SetVariable>
-                <SetVariable name="ConfigKeyRight"><Variable name="group"/>,beatsync_tempo</SetVariable>
+                <SetVariable name="ConfigKeyRight"><Variable name="group"/>,sync_master</SetVariable>
```
Comment (Holzhaus) [inline res/skins/LateNight/deck_rate_controls.xml:193]: This slipped in by mistake?

**Comment by daschuer** [comment]
I think it can be surprising that we disable the explicit master because of pause. 
In general changing unrelated buttons (Sync) due to another button (Play) should be reduced to a minimum.  
I would prefer to still use the unity bpm in this case. 
Simultaneously scratching and even halting the decks will be still possible as long both tracks are in the play state.   

What do you think?

**Comment by ywwg** [comment]
>  I would prefer to still use the unity bpm in this case.

how?  That form of master is not currently supported.

File: res/skins/LateNight/deck_rate_controls.xml (hunk 1 above)
Comment (ywwg) [reply res/skins/LateNight/deck_rate_controls.xml:193]: how else can I test? :)  will remove.

**Comment by ywwg** [comment]
(The key bit of code is in bpmcontrol.cpp calcSyncedRate -- we use the instantaneous bpm from the master.  We could use the base bpm but bpmcontrol doesn't know about it, so it would take some time to develop an alternative.  Unless you have another way in mind)

**Comment by ywwg** [comment]
I do think you're right that not allowing explicit master is weird.  But without a beatdistance to match to, I'm not sure how this works

**Comment by daschuer** [comment]
The internal master is already such a master. It has as BPM value but is not playing. I think we can treat other stopped explicite master's the same. 

No idea if thay is a big effort of not.

**Comment by ywwg** [comment]
but internal clock never stops running, so the beat distance is always updating.  If you've got deck 1 in explicit master and stopped, and deck 2 synced to it and quantize is on, what does it quantize to?

**Comment by ywwg** [comment]
moved to milestone 2.4.0

**Comment by Holzhaus** [comment]
> moved to milestone 2.4.0

Why? Also this still targets the 2.3 branch.

**Comment by ywwg** [comment]
We are not enabling explicit master for 2.3 because there are too many difficult corner cases to fix.  For 2.3, we will keep the current behavior as 2.2, where we only support auto master.  I hope to get everything working for 2.4.  (Yes I need to retarget the PR)

**Comment by Be-ing** [comment]
Is this ready for review?

**Comment by Holzhaus** [comment]
@ywwg Are you planning to rebase this on top of #3698? That PR does allow setting explicit leader on stopped decks.

**Comment by ywwg** [comment]
oh this is obsolete

**Pull request closed without merging**